### ISSUE TITLE
[Feature Request] A way to set an initial maximum window size.

### ISSUE THREAD

**Issue by EzequielBruni** [glzr-io/glazewm#806]
### Describe the problem/motivation

I have a wide monitor. As in 5120x1440p, 32:9. When I have any single window open, I really don't want to have it cover my full screen. Is there any way to set an initial maximum window size?

### Describe the solution you'd like

A simple setting in the `config.yaml` file to have an initial maximum width and height for new windows, which can then be resized manually as needed. These initial maximum dimensions should only be applied when there are no other windows open, or as long as there is empty space left on the screen. Widths and heights could be set by pixel dimensions or percentages.

### Alternatives considered

I could use per-app rules in AltSnap to enforce the max-width on a per-app basis, I suppose, but I don't actually know how it would work with GlazeWM and AltSnap essentially fighting over my window sizes.

**Comment by DreamMaoMao** [comment]
I think the screen state should occupy the entire window, according to the wm rules under linux, it is generally allowed to set the window rules to specify the initial size of the floating window

**Comment by EzequielBruni** [comment]
> I think the tiling state should occupy the entire window, according to the wm rules under linux, it is generally allowed to set the window rules to specify the initial size of the floating window

That's all well and good until you open up an app or page that doesn't have a dark mode for some reason and get flashbanged by your own monitor. And this isn't Linux. And isn't Linux supposed to be about choice, in any case? I'm not asking for this as a philosopher, or expert on wm rules. I just want the choice of not having every first window go automatically full screen on a monitor where that actually hurts the experience.

If y'all can't make that happen, just decide not to, or it's not a priority, that's fair. Y'all are making this for free, and I make no demands. But it seems silly to let an arbitrary rule like that get in the way of better UX, even if my situation is admittedly an edge case.

**Comment by DreamMaoMao** [comment]
I don't think you understand the meaning of tiling, and this is not mandatory, you said that the description of things are so-called floating mode, you can not use tiling mode, and then set all the Windows to the initial state of floating, and then preset the size of the floating window

**Comment by DreamMaoMao** [comment]
You have to use both tiling mode and a preset window size, which can be quite confusing

**Comment by ShortArrow** [comment]
@DreamMaoMao @EzequielBruni It seems you're confusing fullscreen as a tiling window with fullscreen as a floating window.
